Add columns and ActiveModel::Validations support

genail opened the pull request:
By including ActiveModel::Validations in the item,
ActiveRecord-like validations can be used when working with
Dynomite.

To get this working, I needed to implement a `column` method that
allows to define what columns *should* exist in every model.
Without it, ActiveModel::Validations couldn't tell if missing data
is a developer or user mistake.

genail commented:
I couldn't find a way to actually create a test ActiveModel::Validations integration since it's not a dependency of dynomite. @tongueroo do you have an idea how this can be done?

tongueroo commented:
> RE: I couldn't find a way to actually create a test ActiveModel::Validations integration since it's not a dependency of dynomite. @tongueroo do you have an idea how this can be done?

@genail I do actually. I've used this trick and add it to Gemfile under the development, test group. Something like this to the [Gemfile](https://github.com/tongueroo/dynomite/blob/298eaa88a97fe1eee514f6a1eb78c6c741bfc270/Gemfile):

```
source "https://rubygems.org"
git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
gemspec

# only for specs - add this
group :development, :test do
  gem "activemodel"
end
```

Think it's fine to add `activemodel` to the gem dependency though.

Also, I forgot to add to the README that the id automatically gets generated a unique id if it is not set. 

https://github.com/tongueroo/dynomite/blob/298eaa88a97fe1eee514f6a1eb78c6c741bfc270/lib/dynomite/item.rb#L203

tongueroo commented:
Btw. Looks great so let me know when you think it’s ready 👍

genail commented:
@tongueroo I've updated my PR with f66df3b.

- Add ActiveModel::Validations (group=test,development) dependency
- Add ActiveModel::Validations Item integration spec

Also, I've updated the readme by removing a redundant post.id assignment and I've made my previous descriptions a little more readable in my opinion.

genail commented:
Please hold on...

tongueroo commented:
Released in v1.2.0 [CHANGELOG](https://github.com/tongueroo/dynomite/blob/master/CHANGELOG.md) 🎉

Played with it a bit it works great. Understand why you're thinking about method_missing now. See that you ended up using define_method. Either method_missing or define_method works 😁 It's very nice that we can access the column attributes via ruby-ish getter methods now:

```ruby
    >> p = Post.scan.first
    I, [2019-01-12T18:35:15.372784 #11580]  INFO -- : It's recommended to not use scan for production. It can be slow and expensive. You can a LSI or GSI and query the index instead.
    I, [2019-01-12T18:35:15.373974 #11580]  INFO -- : Scanning table: demo-dev-posts
    => #<Post:0x000055a1adabaf28 @attrs={"created_at"=>"2019-01-10T14:54:19Z", "id"=>"390b66398e198ae80e7a1bc4b933db4b81b7c340", "title"=>"test title", "updated_at"=>"2019-01-10T14:54:19Z"}>
    >> p.title # sweetness added by the column declaration
    => "test title"
    >> p.attrs['created_at'] # normal access
    => "2019-01-10T14:54:19Z"
    >> 
```


Thanks so much! 😄
